base-containers/base/inginious_container_api/run_student.py

```diff
@@ -109,7 +112,19 @@ def receive_signal(signum_s):
 
             signal_handler_callback(receive_signal)
 
+        if ssh:  # The student_container will send id and password for ssh connection, transfer it to the agent
+            ssh_id = msgpack.loads(connection.recv(250))  # TODO: Why 250? It works but why ?
```

Comment:
You have to ensure that the message length is < 250 bytes

Reply:
Now:
The message sent is: message = {"type": "ssh_student", "ssh_user": ssh_user_value, "password": password}
It has a constant size of 240 bytes (indeed sys.getsizeof(message)==240)
The password size is of constant size and users are stored as 0 for worker, 1 for root.

Thus, now I use: ssh_id = msgpack.loads(connection.recv(240))
I hope it resolves the potential issues here :)